setup_resources.sh

```diff
@@ -0,0 +1,157 @@
+
+# Wait for the cluster deletion to complete
+echo "Waiting for the cluster to be deleted..."
+while az aro show --name $CLUSTER_NAME --resource-group $CLUSTER_RESOURCEGROUP &> /dev/null; do
+  echo "Cluster is still being deleted...waiting 30 seconds."
+  sleep 30
+done
+
+# Create resource group
+echo "Creating resource group $CLUSTER_RESOURCEGROUP in $LOCATION..."
+az group create --name $CLUSTER_RESOURCEGROUP --location $LOCATION
+
+# Create virtual network
+echo "Creating virtual network $CLUSTER_VNET in $CLUSTER_RESOURCEGROUP..."
+az network vnet create --resource-group $CLUSTER_RESOURCEGROUP --name $CLUSTER_VNET --address-prefixes 10.0.0.0/22
+
+# Delete any existing subnets and associated resources
+echo "Deleting any existing master subnet resources..."
+az network vnet subnet delete --resource-group $CLUSTER_RESOURCEGROUP --vnet-name $CLUSTER_VNET --name $CLUSTER_MASTER_SUBNET || true
+
+echo "Deleting any existing worker subnet resources..."
+az network vnet subnet delete --resource-group $CLUSTER_RESOURCEGROUP --vnet-name $CLUSTER_VNET --name $CLUSTER_WORKER_SUBNET || true
+
+# Create master subnet
+echo "Creating master subnet $CLUSTER_MASTER_SUBNET in $CLUSTER_VNET..."
+az network vnet subnet create --resource-group $CLUSTER_RESOURCEGROUP --vnet-name $CLUSTER_VNET --name $CLUSTER_MASTER_SUBNET --address-prefixes 10.0.0.0/23 --service-endpoints Microsoft.ContainerRegistry
+
+# Create worker subnet
+echo "Creating worker subnet $CLUSTER_WORKER_SUBNET in $CLUSTER_VNET..."
+az network vnet subnet create --resource-group $CLUSTER_RESOURCEGROUP --vnet-name $CLUSTER_VNET --name $CLUSTER_WORKER_SUBNET --address-prefixes 10.0.2.0/23 --service-endpoints Microsoft.ContainerRegistry
+
+# Create cluster
+echo "Creating cluster $CLUSTER_NAME in $CLUSTER_RESOURCEGROUP..."
+az aro create --resource-group $CLUSTER_RESOURCEGROUP --name $CLUSTER_NAME --vnet $CLUSTER_VNET --master-subnet $CLUSTER_MASTER_SUBNET --worker-subnet $CLUSTER_WORKER_SUBNET --pull-secret "$PULL_SECRET" --location $LOCATION --version $OPENSHIFT_VERSION || {
+  echo "Cluster creation failed. Fetching deployment logs..."
+
+  # Fetch the deployment logs for further analysis
+  deployment_name=$(az deployment group list --resource-group $CLUSTER_RESOURCEGROUP --query '[0].name' -o tsv)
+  if [ -n "$deployment_name" ]; then
+    az deployment group show --name $deployment_name --resource-group $CLUSTER_RESOURCEGROUP
+  else
+    echo "No deployment found for resource group $CLUSTER_RESOURCEGROUP."
+  fi
+
+  exit 1
+}
+
+# Check for the existence of the cluster
+if az aro show --name $CLUSTER_NAME --resource-group $CLUSTER_RESOURCEGROUP &> /dev/null; then
+  echo "Cluster creation successful."
+else
+  echo "Cluster creation failed. Please check the logs for more details."
+  exit 1
+fi
+
+echo "To list cluster credentials, run:"
+echo "    az aro list-credentials --name $CLUSTER_NAME --resource-group $CLUSTER_RESOURCEGROUP"
+
+# Validate RP running
+validate_rp_running
```

Comment:
What's the point of validating that the RP is running down here after doing everything else? Shouldn't we validate that the RP is running before creating the cluster rather than after?